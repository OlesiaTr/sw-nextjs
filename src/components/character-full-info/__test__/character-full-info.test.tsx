import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

import CharacterFullInfo from '../character-full-info';
import type { Character } from '@/helpers';

describe('CharacterFullInfo', () => {
  const MOCK_FUNCTION = jest.fn();
  const MOCK_CHARACTER: Character = {
    name: 'Luke Skywalker',
    gender: 'male',
    birth_year: '19BBY',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    homeworld: 20,
    id: 1,
    created: '2014-12-10T16:16:29.192000Z',
    edited: '2014-12-20T21:17:50.325000Z',
    films: [1, 2, 3, 4, 5, 6],
    species: [1],
    starships: [48, 59, 64, 65, 74],
    url: 'https://sw-api.starnavi.io/people/1/',
    vehicles: [38],
  };

  const renderCharacterFullInfo = () =>
    render(
      <CharacterFullInfo
        character={MOCK_CHARACTER}
        onCharacterClick={MOCK_FUNCTION}
      />,
    );

  it('renders character details correctly', () => {
    const { asFragment } = renderCharacterFullInfo();

    expect(screen.getByText(MOCK_CHARACTER.birth_year)).toBeInTheDocument();
    expect(screen.getByText(MOCK_CHARACTER.height + ' cm')).toBeInTheDocument();
    expect(screen.getByText(MOCK_CHARACTER.mass + ' kg')).toBeInTheDocument();
    expect(screen.getByText(MOCK_CHARACTER.hair_color)).toBeInTheDocument();
    expect(screen.getByText(MOCK_CHARACTER.skin_color)).toBeInTheDocument();
    expect(screen.getByText(MOCK_CHARACTER.eye_color)).toBeInTheDocument();
    expect(
      screen.getByText('Homeworld: ' + MOCK_CHARACTER.homeworld),
    ).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
