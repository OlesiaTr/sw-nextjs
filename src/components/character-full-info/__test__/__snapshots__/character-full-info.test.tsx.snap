// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterFullInfo renders character details correctly 1`] = `
<DocumentFragment>
  <div
    class="bg-mainGray py-1 px-3 rounded-sm border-borderGray flex flex-col items-start"
  >
    <p
      class="text-2xl font-bold mb-3"
    >
      Homeworld: 20
    </p>
    <div
      class="flex flex-wrap justify-between mb-3 font-medium "
    >
      <p
        class="w-1/2"
      >
        <span
          class="font-semibold"
        >
          Birth Year:
        </span>
         19BBY
      </p>
      <p
        class="w-1/2"
      >
        <span
          class="font-semibold"
        >
          Height:
        </span>
         172 cm
      </p>
      <p
        class="w-1/2"
      >
        <span
          class="font-semibold"
        >
          Mass:
        </span>
         77 kg
      </p>
      <p
        class="w-1/2"
      >
        <span
          class="font-semibold"
        >
          Hair Color:
        </span>
         blond
      </p>
      <p
        class="w-1/2"
      >
        <span
          class="font-semibold"
        >
          Skin Color:
        </span>
         fair
      </p>
      <p
        class="w-1/2"
      >
        <span
          class="font-semibold"
        >
          Eye Color:
        </span>
         blue
      </p>
    </div>
  </div>
</DocumentFragment>
`;
